dot/sync/worker_test.go

```diff
@@ -32,30 +40,47 @@ func TestWorker(t *testing.T) {
 	w := newWorker(peerA, sharedGuard, reqMaker)
 
 	wg := sync.WaitGroup{}
-	queue := make(chan *syncTask, 2)
+	queue := make(chan *syncTask, 1) // 1 means that maximum 1 worker can send request at a given time
```

Comment:
here we want a buffered channel with size 2 so we can dispatch 2 messages in a row for the workers